.github/workflows/slow-tests.yml

```diff
@@ -60,14 +60,19 @@ jobs:
       - uses: ./.github/actions/free-disk-space
       - uses: ./.github/actions/untracked
       - uses: ./.github/actions/setup-python
+        with:
+          python-version: 3.9
       - uses: ./.github/actions/setup-pyenv
       - uses: ./.github/actions/setup-java
       - name: Install dependencies
         run: |
           python -m venv .venv
           source .venv/bin/activate
           source ./dev/install-common-deps.sh --ml
-          python -m pip install langchain_experimental
+          # We need to install `tf-keras` because the transformer flavor
+          # tests depends on Keras < 3 but the `./dev/install-common-deps.sh`
+          # installs Keras 3 in python 3.9 environment.
+          python -m pip install langchain_experimental tf-keras==2.*
```

Comment:
```suggestion
          # We need to install `tf-keras` because the transformer flavor
          # tests depends on Keras < 3 but the `./dev/install-common-deps.sh`
          # installs Keras 3 in python 3.9 environment.
          python -m pip install langchain_experimental keras<3
```

Reply:
Strange error happens after this change:
https://github.com/mlflow/mlflow/actions/runs/9658159962/job/26638787225?pr=12468#step:8:2055

I think installing `tf-keras` is better, `tf-keras` is equivalent to `keras<=2` and `tf-keras` lib can co-exist with `keras 3`

Reply:
@WeichenXu123 I think that error is transient. `c2f8398e-0208-48f9-865e-72ac3d91a5ea.sh` is a temporary shell script created by github to run the step. You can rerun the job.

Reply:
```
% docker run --rm python:3.9 pip install tf-keras
Collecting tf-keras
  Downloading tf_keras-2.16.0-py3-none-any.whl (1.7 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.7/1.7 MB 8.9 MB/s eta 0:00:00
Collecting tensorflow<2.17,>=2.16
  Downloading tensorflow-2.16.1-cp39-cp39-manylinux_2_17_aarch64.manylinux2014_aarch64.whl (218.9 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 218.9/218.9 MB 4.5 MB/s eta 0:00:00
Collecting google-pasta>=0.1.1
  Downloading google_pasta-0.2.0-py3-none-any.whl (57 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 57.5/57.5 kB 9.1 MB/s eta 0:00:00
Requirement already satisfied: setuptools in /usr/local/lib/python3.9/site-packages (from tensorflow<2.17,>=2.16->tf-keras) (58.1.0)
Collecting tensorflow-io-gcs-filesystem>=0.23.1
  Downloading tensorflow_io_gcs_filesystem-0.37.0-cp39-cp39-manylinux_2_17_aarch64.manylinux2014_aarch64.whl (4.8 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 4.8/4.8 MB 10.2 MB/s eta 0:00:00
Collecting requests<3,>=2.21.0
  Downloading requests-2.32.3-py3-none-any.whl (64 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 64.9/64.9 kB 7.2 MB/s eta 0:00:00
Collecting packaging
  Downloading packaging-24.1-py3-none-any.whl (53 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 54.0/54.0 kB 7.1 MB/s eta 0:00:00
Collecting opt-einsum>=2.3.2
  Downloading opt_einsum-3.3.0-py3-none-any.whl (65 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 65.5/65.5 kB 7.3 MB/s eta 0:00:00
Collecting keras>=3.0.0
  Downloading keras-3.4.0-py3-none-any.whl (1.1 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.1/1.1 MB 10.4 MB/s eta 0:00:00
```

Reply:
uh ? why it installs tf-keras 2 here ? https://github.com/WeichenXu123/mlflow/actions/runs/9655317212/job/26631012047#step:8:2135

Reply:
Oh,
I think in your environment, `pip install tf-keras` installs other dependencis that triggers installing `keras3`, but at the same time `tf-keras 2.16` will also be installed,
you can verify it by:
```
importing tf_keras
tf_keras.__version__
```